llvm/lib/MC/ELFObjectWriter.cpp

```diff
@@ -951,6 +965,38 @@ void ELFWriter::writeSectionHeader(const MCAssembler &Asm) {
     else
       Size = Offsets.second - Offsets.first;
 
+    auto SectionHasFlag = [&](uint64_t Flag) -> bool {
+      return Section->getFlags() & Flag;
+    };
+    auto SectionIsType = [&](uint64_t Type) -> bool {
+      return Section->getType() == Type;
+    };
+
+    if (SectionIsType(ELF::SHT_STRTAB)) {
+      stats::StrtabBytes += Size;
+    } else if (SectionIsType(ELF::SHT_SYMTAB)) {
+      stats::SymtabBytes += Size;
+    } else if (SectionIsType(ELF::SHT_DYNSYM)) {
+      stats::DynsymBytes += Size;
+    } else if (SectionIsType(ELF::SHT_REL) || SectionIsType(ELF::SHT_RELA) ||
+               SectionIsType(ELF::SHT_RELR) || SectionIsType(ELF::SHT_CREL)) {
+      stats::RelocationBytes += Size;
+    } else if (SectionIsType(ELF::SHT_X86_64_UNWIND)) {
+      stats::UnwindBytes += Size;
+    } else if (Section->getName().starts_with(".debug")) {
+      stats::DebugBytes += Size;
+    } else if (SectionHasFlag(ELF::SHF_ALLOC)) {
```

Comment:
Perhaps check SHF_ALLOC first, then .debug, then use a switch? Then, `SectionIsType` could be removed.

Reply:
I moved the check for eh_frame first since it's also SHF_ALLOC

Reply:
`== ".eh_frame"`

The design makes it monolithic.